Monika After Story/game/zz_windowreacts.rpy

```diff
@@ -0,0 +1,343 @@
+        return persistent._mas_enable_notifications and persistent._mas_windowreacts_notif_filters.get(group,False)
+
+    def mas_tryShowNotificationOSX(title, body):
+        """
+        Tries to push a notification to the notification center on macOS.
+        If it can't it should fail silently to the user.
+
+        IN:
+            title: notification title
+            body: notification body
+        """
+        os.system('osascript -e \'display notification "{0}" with title "{1}"\''.format(body,title))
+    
+    def mas_tryShowNotificationLinux(title, body):
+        """
+        Tries to push a notification to the notification center on Linux.
+        If it can't it should fail silently to the user.
+
+        IN:
+            title: notification title
+            body: notification body
+        """
+        os.system("notify-send '{0}' '{1}' -u low".format(title,body))
+
+
+#Notification creation label
+#IN:
+#   title: Notification heading text
+#   body: Notification body text
+#   group: Notification group (for checking if we have this enabled)
+#   skip_checks: Whether or not we skips checks
+
+label display_notif(title, body, group=None, skip_checks=False):
+    #We only show notifications if:
+    #We are able to show notifs
+    #MAS isn't the active window
+    #User allows them
+    #And if the notification group is enabled
+
+    #OR if we skip checks
+    #NOTE: THIS IS TO ONLY BE USED FOR INTRODUCTORY PURPOSES
+
+    #First we want to create this location in the dict, but don't add an extra location if we're skipping checks
+    if persistent._mas_windowreacts_notif_filters.get(group) is None and not skip_checks:
+        $ persistent._mas_windowreacts_notif_filters[group] = False
+
+    if (
+            (
+                mas_windowreacts.can_show_notifs
+                and ((renpy.windows and not mas_isFocused()) or not renpy.windows)
+                and mas_notifsEnabledForGroup(group)
+            )
+            or skip_checks
+        ):
+
+        #Play the notif sound
+        play sound "mod_assets/sounds/effects/notif.wav"
+
+        #Now we make the notif
+        if (renpy.windows):
+            # The Windows way
+            $ tip.showWindow(renpy.substitute(title),renpy.substitute(body))
+
+            #We need the IDs of the notifs to delete them from the tray
+            $ destroy_list.append(tip.hwnd)
+
+        elif (renpy.macintosh):
+            # The macOS way
+            $ mas_tryShowNotificationOSX(renpy.substitute(title),renpy.substitute(body))
+
+        elif (renpy.linux):
+            # The Linux way
+            $ mas_tryShowNotificationLinux(renpy.substitute(title),renpy.substitute(body))
+    return
+
+
+#START: Window Reacts
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_windowreacts_database,
+            eventlabel="monika_whatwatching",
+            category=['youtube'],
+            conditional="mas_notifsEnabledForGroup('Window Reactions')",
```

Comment:
I see what this is trying to do, but I think this could be better handled as something in the rule property instead. maybe like:
`rules={"winreact": <string name of window react group>}`

That way we dont always have to add `mas_notifsEnabledForGroup` to every window react topic.